match/lib/match/git_helper.rb

```diff
@@ -59,6 +60,16 @@ def self.clone(git_url,
 
       checkout_branch(branch) unless branch == "master"
 
+      # makes it backwards compatible allowing checkout and decryption of repo
+      # if open ssl version file not exist
+      openssl_version_file = File.join(@dir, OPENSSL_VERSION_FILE_NAME)
+      if File.exist?(openssl_version_file)
+        repo_openssl_version = File.read(openssl_version_file)
+        if repo_openssl_version != Encrypt.openssl_version.to_s
+          UI.user_error!("Repository was encrypted using '#{repo_openssl_version}' and currently used version is '#{Encrypt.openssl_version}'")
```

Comment:
Is it possible that decrypting with different OpenSSL versions _may_ still work though? I have a small fear that this may error out on the user for a minor/patch OpenSSL version where decrypting may still work resulting in a false failure. How would a user handle this error when it occurs?

Which makes me think this may be better but after a decrypt fails **but** then it may not be due to OpenSSL versions but could also be due to password 🤔

Would it be better for us to store the method of encrypting instead of storing the OpenSSL version?

Reply:
I’m not sure if it’s possible that two different OpenSSL versions could produce different hashes, probably not likely. Not sure about OpenSSL and LibreSSL producing different hashes though.

Method of encryption does not change that frequently (i guess?). During the transiton it’s useful info, but in general not really